Languages/IronPython/IronPython/Hosting/PythonOptionsParser.cs

```diff
@@ -138,7 +138,16 @@ public sealed class PythonOptionsParser : OptionsParser<PythonConsoleOptions> {
                     LanguageSetup.Options["Arguments"] = PopRemainingArgs();
                     break;
 
+                case "-X:NoFrames":
+                    if(LanguageSetup.Options.ContainsKey("Frames") && LanguageSetup.Options["Frames"] != ScriptingRuntimeHelpers.False) {
+                        throw new InvalidOptionException("Only one of -X:Frames/-X:NoFrames may be specified");
+                    }
+                    LanguageSetup.Options["Frames"] = ScriptingRuntimeHelpers.False;
+                    break;
                 case "-X:Frames":
+                    if (LanguageSetup.Options.ContainsKey("Frames") && LanguageSetup.Options["Frames"] != ScriptingRuntimeHelpers.True) {
```

Comment:
FullFrames also turns on Frames, maybe stick this check in the -X:FullFrames case as well?